File: AndroidMVVMJsonExample/app/src/main/java/com/blackbirds/shakil/androidmvvmjsonexample/ViewModel/MainActivityViewModel.java
package com.blackbirds.shakil.androidmvvmjsonexample.ViewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.blackbirds.shakil.androidmvvmjsonexample.Model.MovieModel;
import com.blackbirds.shakil.androidmvvmjsonexample.Repositories.MainRepo;

import java.util.List;

public class MainActivityViewModel extends ViewModel {
    private final MainRepo mainRepo;

    public MainActivityViewModel() {
        mainRepo = new MainRepo();
    }

    public LiveData<List<MovieModel>> getMovieList(){
        return mainRepo.getMovieModelLiveData();
    }
}

File: KotlinMVVMJsonExample/app/src/main/java/com/blackbirds/shakil/kotlinmvvmjsonexample/ViewModel/MainActivityViewModel.kt
package com.blackbirds.shakil.kotlinmvvmjsonexample.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.blackbirds.shakil.kotlinmvvmjsonexample.Model.MovieModel
import com.blackbirds.shakil.kotlinmvvmjsonexample.Repositories.MainRepo

class MainActivityViewModel : ViewModel() {
    private val mainRepo: MainRepo
    val getMovieList: LiveData<MutableList<MovieModel>>
        get() = mainRepo.getMovieModelLiveData

    init {
        mainRepo = MainRepo()
    }
}